src/multisig/multisigTransactionsFactory.ts

```diff
@@ -0,0 +1,509 @@
+import {
+    AddressValue,
+    ArgSerializer,
+    BigUIntValue,
+    ContractFunction,
+    EndpointDefinition,
+    EndpointModifiers,
+    NativeSerializer,
+    OptionType,
+    OptionValue,
+    TokenIdentifierValue,
+    U32Value,
+    U64Type,
+    U64Value,
+    VariadicValue,
+} from "../abi";
+import { TokenComputer, TransactionsFactoryConfig } from "../core";
+import { Address } from "../core/address";
+import { Transaction } from "../core/transaction";
+import { TransactionBuilder } from "../core/transactionBuilder";
+import { SmartContractTransactionsFactory } from "../smartContracts";
+import * as resources from "./resources";
+
+interface IAbi {
+    constructorDefinition: EndpointDefinition;
+    upgradeConstructorDefinition?: EndpointDefinition;
+
+    getEndpoint(name: string | ContractFunction): EndpointDefinition;
+}
+/**
+ * Use this class to create multisig related transactions like creating a new multisig contract,
+ * proposing actions, signing actions, and performing actions.
+ */
+export class MultisigTransactionsFactory extends SmartContractTransactionsFactory {
+    private readonly argSerializer: ArgSerializer;
+
+    constructor(options: { config: TransactionsFactoryConfig; abi?: IAbi }) {
+        super(options);
+        this.argSerializer = new ArgSerializer();
+    }
+
+    /**
+     * Creates a transaction to deploy a new multisig contract
+     */
+    createTransactionForMultisigDeploy(sender: Address, options: resources.DeployMultisigContractInput): Transaction {
+        const nativeTransferAmount = options.amount ?? 0n;
+        const boardAddresses: AddressValue[] = options.board.map((addr) => new AddressValue(addr));
+        const args = [new U32Value(options.quorum), VariadicValue.fromItems(...boardAddresses)];
+
+        return this.createTransactionForDeploy(sender, {
+            bytecode: options.bytecode,
+            gasLimit: options.gasLimit,
+            nativeTransferAmount,
+            isUpgradeable: options.isUpgradeable,
+            isReadable: options.isReadable,
+            isPayable: options.isPayable,
+            isPayableBySmartContract: options.isPayableBySmartContract,
+            arguments: args,
+        });
+    }
+
+    /**
+     * Proposes adding a new board member
+     */
+    createTransactionForProposeAddBoardMember(
+        sender: Address,
+        options: resources.ProposeAddBoardMemberInput,
+    ): Transaction {
+        const dataParts = [
+            "proposeAddBoardMember",
+            this.argSerializer.valuesToStrings([new AddressValue(options.boardMemberAddress)])[0],
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: sender,
+            receiver: options.multisigContract,
+            dataParts: dataParts,
+            gasLimit: options.gasLimit,
+            addDataMovementGas: true,
```

Comment:
I think we should not add extra gas limit. Since the gas is provided by the user, he should consider everything. Same thing for the methods bellow.